os/fs/smartfs/smartfs_utils.c

```diff
@@ -521,6 +521,253 @@ int set_used_byte_count(uint8_t *used, uint16_t count)
+#endif
+		} else {
+#ifdef CONFIG_SMARTFS_DYNAMIC_HEADER
+			set_used_byte_count((uint8_t *)header->used, used_value + sf->byteswritten);
+#else
+			*((uint16_t *)header->used) += sf->byteswritten;
+#endif
+		}
+
+		/* Write the entire sector to FLASH */
+
+		smartfs_setbuffer(&readwrite, sf->currsector, 0, fs->fs_llformat.availbytes, (uint8_t *)sf->buffer);
+		ret = FS_IOCTL(fs, BIOC_WRITESECT, (unsigned long)&readwrite);
+		if (ret < 0) {
+			fdbg("Error %d writing used bytes for sector %d\n", ret, sf->currsector);
+			goto errout;
+		}
+
+		sf->byteswritten = 0;
+		/* File's data sector has been synced with MTD, now check if this is a new file entry and write the entry */
+		if (sf->bflags & SMARTFS_BFLAG_NEW_ENTRY) {
+			/* Flags for this entry have already been set and stored, so mode will not be used */
+			ret = smartfs_writeentry(fs, sf->entry, SMARTFS_DIRENT_TYPE_FILE, (sf->entry.flags & SMARTFS_DIRENT_MODE));
+			if (ret < 0) {
+				fdbg("Failed to write new file entry ot MTD\n");
+				goto errout;
+			}
+		}
+		sf->bflags = SMARTFS_BFLAG_UNMOD;
+	}
+#else							/* CONFIG_SMARTFS_USE_SECTOR_BUFFER */
+
+	/* Test if we have written bytes to the current sector that
+	 * need to be recorded in the chain header's used bytes field. */
+
+	if (sf->byteswritten > 0) {
+		fvdbg("Syncing sector %d\n", sf->currsector);
+
+		/* Read the existing sector used bytes value */
+
+		header = (struct smartfs_chain_header_s *)fs->fs_rwbuffer;
+		smartfs_setbuffer(&readwrite, sf->currsector, 0, sizeof(struct smartfs_chain_header_s), (uint8_t *)fs->fs_rwbuffer);
+		ret = FS_IOCTL(fs, BIOC_READSECT, (unsigned long)&readwrite);
+		if (ret < 0) {
+			fdbg("Error %d reading sector %d data\n", ret, sf->currsector);
+			goto errout;
+		}
+#ifdef CONFIG_SMARTFS_DYNAMIC_HEADER
+		used_value = sf->entry.datlen % (fs->fs_llformat.availbytes - sizeof(struct smartfs_chain_header_s));
+		if (sf->entry.datlen > 0 && used_value == 0) {
+			used_value = fs->fs_llformat.availbytes - sizeof(struct smartfs_chain_header_s);
+		}
+
+		set_used_byte_count((uint8_t *)header->used, used_value);
+#else
+		if (SMARTFS_USED(header) == SMARTFS_ERASEDSTATE_16BIT) {
+			header->used[0] = (uint8_t)(sf->byteswritten & 0x00FF);
+			header->used[1] = (uint8_t)(sf->byteswritten >> 8);
+		} else {
+			uint16_t tmp = SMARTFS_USED(header);
+			tmp += sf->byteswritten;
+			header->used[0] = (uint8_t)(tmp & 0x00FF);
+			header->used[1] = (uint8_t)(tmp >> 8);
+		}
+#endif
+		smartfs_setbuffer(&readwrite, sf->currsector, offsetof(struct smartfs_chain_header_s, used), sizeof(uint16_t), (uint8_t *)&fs->fs_rwbuffer[offsetof(struct smartfs_chain_header_s, used)]);
+		ret = FS_IOCTL(fs, BIOC_WRITESECT, (unsigned long)&readwrite);
+		if (ret < 0) {
+			fdbg("Error %d writing used bytes for sector %d\n", ret, sf->currsector);
+			goto errout;
+		}
+
+		sf->byteswritten = 0;
+	}
+#endif							/* CONFIG_SMARTFS_USE_SECTOR_BUFFER */
+
+errout:
+	return ret;
+}
+
```

Comment:
can you also remove this?

Reply:
Always a line gap after the ending of a function.